core/templates/pages/classroom-admin-page/classroom-admin-page.component.ts

```diff
@@ -761,7 +770,21 @@ export class ClassroomAdminPageComponent implements OnInit {
     );
   }
 
-  saveClassroomValidationErrors(): string[] {
+  canSaveClassroom(): boolean {
+    if (
+      (this.tempClassroomData.getIsPublished() &&
+        this.validationErrors.length !== 0) ||
+      this.saveClassroomValidationErrors.length !== 0
+    ) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
I think this function would be clearer like this
```suggestion
  canSaveClassroom(): boolean {
    if (this.tempClassroomData.getIsPublished()) {
      return this.validationErrors.length === 0
    }
    return this.saveClassroomValidationErrors.length === 0
  }
```